pkg/forwarder/sync_forwarder.go

```diff
@@ -18,12 +19,15 @@ type SyncForwarder struct {
 	client           *http.Client
 }
 
+var forwarderTimeout = config.Datadog.GetDuration("forwarder_timeout") * time.Second
```

Comment:
I don't think this is a good idea to move this here, it is better to read that config field from the main than in a global variable:
 1. It's worth having it in the constructor signature `NewSyncForwarder`, to let the user of this constructor know that there is a timeout involved
 2. I'm not entirely sure about the init order for global variables and your var here may use an uninitialized `config.Datadog` (even if I think it's fine in this case, it's not great)

In order to have it available in `SyncForwarder.sendHTTPTransactions` to recreate the client, please store the timeout value in the `SyncForwarder` struct by adding a `time.Duration` field.